cirq-core/cirq/devices/line_qubit.py

```diff
@@ -191,7 +191,6 @@ def __new__(cls, x: int, dimension: int) -> 'cirq.LineQid':
             dimension: The dimension of the qid's Hilbert space, i.e.
                 the number of quantum levels.
         """
-        x = int(x)
```

Comment:
I wonder if we should leave this coercion on for integral arguments.
As it is `cirq.LineQubit(np.int8(120))` gets cached and becomes immortal, potentially causing overflows:
```python
In [1]: cirq.LineQubit(np.int8(120))
Out[1]: cirq.LineQubit(120)
In [2]: q120 = cirq.LineQubit(120)
In [3]: q120 + q120
/home/user/Code/gh/quantumlib/Cirq/cirq-core/cirq/devices/line_qubit.py:127: RuntimeWarning: overflow encountered in scalar add
  return self._with_x(x=self._x + other._x)
Out[3]: cirq.LineQubit(-16)
```

Reply:
I'm reluctant to assume any implicit coercion is safe at this point. If someone is using `int8` then presumably they have a reason to be doing so. If we want to do something different here, I think it should go in a separate PR.